rand_core/src/lib.rs

```diff
@@ -42,16 +42,16 @@
 #[cfg(feature = "std")] extern crate std;
 #[cfg(feature = "alloc")] use alloc::boxed::Box;
 
-pub use error::Error;
-#[cfg(feature = "getrandom")] pub use os::OsRng;
-
+use core::{convert::Infallible, fmt};
 
 pub mod block;
-mod error;
 pub mod impls;
 pub mod le;
 #[cfg(feature = "getrandom")] mod os;
 
+#[cfg(feature = "getrandom")] pub use getrandom;
+#[cfg(feature = "getrandom")] pub use os::OsRng;
```

Comment:
I suspect most people (including myself) will be happy to see `rand_core::Error` gone, but I wonder if it will cause problems (for an `alloc`-less `no_std` crate which needs to handle errors).

Reply:
I don't think it should. `getrandom::Error` is a `no_std` type by default. Defining an IO RNG which uses `io::Error` would make it automatically incompatible with `no_std`. So everything will depend on implementer crates.